### PR TITLE
update zkEVM - add endpoints

### PR DESCRIPTION
**Pull request by EmpieichO**
Update zkEVM - add endpoints ... in response to the request here https://0xpolygon.slack.com/archives/C061ADK8534/p1705070519246269

### CODE REVIEW

**Review by 0xgraciegrace — APPROVED** [review]

File: docs/zkEVM/architecture/zknode/index.md [hunk 1]
```diff
@@ -20,3 +20,56 @@ Polygon zkEVM achieves this by utilizing several actors. Here is a list of the m
...
+### RPC endpoints
+
+Any user can participate in this role, as an RPC node.
+
+Required services and components:
+
+- JSON RPC: can run in a separated instance, and can have multiple instances
+- Synchronizer: single instance that can run on a separate instance
+- Executor & Merkletree: service that can run on a separate instance
+- State DB: Postgres SQL that can be run in a separate instance
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/zknode/index.md:37]:
```suggestion
- JSON RPC: can run on a separated instance, and can have multiple instances.
- Synchronizer: single instance that can run on a separate instance.
- Executor & Merkletree: service that can run on a separate instance.
- State DB: Postgres SQL that can run on a separate instance.
```

File: docs/zkEVM/architecture/zknode/index.md [hunk 2]
```diff
@@ -20,3 +20,56 @@ Polygon zkEVM achieves this by utilizing several actors. Here is a list of the m
...
+
+There must be only one synchronizer, and it's recommended that it must have exclusive access to an executor instance, though not necessarily.
+
+ The synchronizer role can be run perfectly in a single instance, but the JSON RPC and executor services can benefit from running in multiple instances, if the performance decreases due to the number of received requests.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/zknode/index.md:41]:
```suggestion
 The synchronizer role can run perfectly in a single instance, but the JSON RPC and executor services can benefit from running in multiple instances, if the performance decreases due to the number of received requests.
```

File: docs/zkEVM/architecture/zknode/index.md [hunk 3]
```diff
@@ -20,3 +20,56 @@ Polygon zkEVM achieves this by utilizing several actors. Here is a list of the m
...
+
+This role can only be performed by a single entity. This is enforced in the smart contract, as the related methods of the trusted sequencer can only be performed by the owner of a particular private key.
+
+Required services and components:
+
+- JSON RPC: can run in a separated instance, and can have multiple instances
+- Sequencer & Synchronizer: single instance that needs to run together
+- Executor & Merkletree: service that can run on a separate instance
+- Pool DB: Postgres SQL that can be run in a separate instance
+- State DB: Postgres SQL that can be run in a separate instance
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/zknode/index.md:56]:
```suggestion
- JSON RPC: can run on a separated instance, and can have multiple instances.
- Sequencer & synchronizer: single instance that needs to run them together.
- Executor & Merkletree: service that can run on a separate instance.
- Pool DB: Postgres SQL that can run on a separate instance.
- State DB: Postgres SQL that can run on a separate instance.
```

File: docs/zkEVM/architecture/zknode/index.md [hunk 4]
```diff
@@ -20,3 +20,56 @@ Polygon zkEVM achieves this by utilizing several actors. Here is a list of the m
...
+
+Note that the JSON RPC is required to receive transactions. It's recommended that the JSON RPC runs on separated instances, and potentially more than one (depending on the load of the network). It's also recommended that the JSON RPC and the Sequencer don't share the same executor instance, to make sure that the sequencer has exclusive access to an executor
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/zknode/index.md:58]:
```suggestion
Note that the JSON RPC is required to receive transactions. It's recommended that the JSON RPC runs on separated instances, and potentially more than one (depending on the load of the network). It's also recommended that the JSON RPC and the sequencer don't share the same executor instance, to make sure that the sequencer has exclusive access to an executor
```

File: docs/zkEVM/architecture/zknode/index.md [hunk 5]
```diff
@@ -20,3 +20,56 @@ Polygon zkEVM achieves this by utilizing several actors. Here is a list of the m
...
+This role can be performed by anyone.
+
+Required services and components:
+
+- Synchronizer: single instance that can run on a separated instance
+- Executor & Merkletree: service that can run on a separate instance
+- State DB: Postgres SQL that can be run in a separate instance
+- Aggregator: single instance that can run on a separated instance
+- Prover: single instance that can run on a separated instance
+- Executor: single instance that can run on a separated instance
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/zknode/index.md:71]:
```suggestion
- Synchronizer: single instance that can run on a separate instance.
- Executor & Merkletree: service that can run on a separate instance.
- State DB: Postgres SQL that can be run on a separate instance.
- Aggregator: single instance that can run on a separate instance.
- Prover: single instance that can run on a separate instance.
- Executor: single instance that can run on a separate instance.
```

**Review by kmurphypolygon — APPROVED** [review]
Made some suggestions.

**Pull request merged**